### PR TITLE
feat(misc): Home cockpit mode

### LINKED ISSUE #6459: Provide a Home Cockpit mode
### Aircraft Version

Development

### Description

Provide an option in the EFB "Home Cockpit Mode" that, when enabled, will provide the following visual changes:

- Remove backlight bleed from PFD, ND, and ECAMs
- Remove reflection from the ISIS

### References (optional)

_No response_

### Additional info (optional)

Cockpit builders may use the built-in "popout" feature for the internal displays by right-alt clicking on the instrument and dragging the resultant popout to an external monitor mounted in various cockpit locations.

These displays often have their own real-world backlight bleed and rendering a software backlight bleed may be undesirable. The ISIS display popout showing reflections can harm the immersion in this context.

### Discord Username (optional)

AntennaRex#9906

### PR DESCRIPTION
**Pull request by ssewell**
Fixes #6459

## Summary of Changes
Provides an option in the EFB as "Home Cockpit Mode" within the Realism tab that, when enabled, provides the following visual changes:
 - Removes backlight bleed from PFD, ND, and ECAMs
 - Removes reflection from the ISIS

## Screenshots (if necessary)

Enabling within the EFB Realism tab:
image

Backlight bleed on displays before:
Before

No backlight bleed on displays after:
After

ISIS Display before (popped out via right-alt click):
Before ISIS

ISIS Display after (popped out via right-alt click):
After ISIS

## References

As this is a visual QoL improvement for cockpit builders, there is no real-world reference available for these changes.

## Additional context

Cockpit builders may use the built-in "popout" feature for the internal displays by right-alt clicking on the instrument and dragging the resultant popout to an external monitor mounted in various cockpit locations.

These displays often have their own real-world backlight bleed and rendering a software backlight bleed may be undesirable. The ISIS display popout showing reflections can harm the immersion in this context.

To remove the ISIS reflections, the current implementation overrides the opacity for this particular instrument. As a result, you cannot change the brightness of this instrument while Home Cockpit Mode is enabled.

Discord username (if different from GitHub): AntennaRex#9906

## Testing instructions

1. View the PFD, ND, or ECAM displays before making any changes. The existing backlight bleed should still be present.
2. View the ISIS, in particular while popped, and change your main view within the cockpit. You should notice a reflection overlay present.
3. In the EFB, open Settings and set Home Cockpit Mode to Enabled under the Realism tab. The backlight bleed should no longer be present in the instruments listed above and the ISIS reflections should be gone.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js [hunk 1]
```diff
@@ -186,5 +189,9 @@ class EICASCommonDisplay extends Airliners.EICASTemplateElement {
         }
         return payloadWeight;
     }
+    refreshHomeCockpitMode() {
+        const isHomeCockpit = SimVar.GetSimVarValue("L:A32NX_HOME_COCKPIT_ENABLED", "bool");
+        this.backlightBleed.style.visibility = (isHomeCockpit === 1) ? "hidden" : "visible";
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js:194]: **issue:** style property is not diffed

File: src/instruments/src/Common/displayUnit.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ export const DisplayUnit: React.FC<DisplayUnitProps> = (props) => {
 
     const [potentiometer] = useSimVar(`LIGHT POTENTIOMETER:${props.potentiometerIndex}`, 'percent over 100', 200);
     const [electricityState] = useSimVar(props.electricitySimvar, 'bool', 200);
+    const [homeCockpit] = useSimVar('L:A32NX_HOME_COCKPIT_ENABLED', 'bool', 200);
```
Comment (Benjozork) [inline src/instruments/src/Common/displayUnit.tsx:29]: **issue:** LVar is not documented in `docs/a320-simvars.md`

File: src/instruments/src/Common/displayUnit.tsx [hunk 3]
```diff
@@ -87,6 +88,12 @@ export const DisplayUnit: React.FC<DisplayUnitProps> = (props) => {
...
             <></>
         );
+    } if (homeCockpit === 1) {
+        return (
+            <>
+                <div style={{ display: state === DisplayUnitState.On ? 'block' : 'none' }}>{props.children}</div>
+            </>
+        );
     }
     return (
```
Comment (Benjozork) [inline src/instruments/src/Common/displayUnit.tsx:98]:
```suggestion
    }

    if (homeCockpit === 1) {
        return (
            <>
                <div style={{ display: state === DisplayUnitState.On ? 'block' : 'none' }}>{props.children}</div>
            </>
        );
    }

    return (
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js (hunk 1 above)
Comment (ErickSharp) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js:194]:
```suggestion
        this.backlightBleed.style.visibility = isHomeCockpit ? "hidden" : "visible";
```

File: src/instruments/src/Common/displayUnit.tsx [hunk 4]
```diff
@@ -87,6 +88,12 @@ export const DisplayUnit: React.FC<DisplayUnitProps> = (props) => {
         return (
             <></>
         );
+    } if (homeCockpit === 1) {
```
Comment (ErickSharp) [inline src/instruments/src/Common/displayUnit.tsx:91]:
```suggestion
    } if (homeCockpit) {
```

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 5]
```diff
@@ -467,6 +473,21 @@ const RealismPage = () => {
                             ))}
                         </SelectGroup>
                     </div>
+
+                    <div className="py-4 flex flex-row justify-between items-center">
+                        <span className="text-lg text-gray-300 mr-1">Home Cockpit Mode</span>
+                        <SelectGroup>
+                            {homeCockpitButtons.map((button) => (
+                                <SelectItem
```
Comment (ErickSharp) [inline src/instruments/src/EFB/Settings/Settings.tsx:481]: This setting is binary in nature -- there is no need to have SelectItems that are only 'Enabled' and 'Disabled'; use the Toggle component here instead.

File: src/instruments/src/ISIS/AutoBrightness.tsx [hunk 6]
```diff
@@ -88,6 +89,14 @@ export const AutoBrightness: React.FC<AutoBrightnessProps> = ({ bugsActive, chil
         { additionalDeps: [currentBrightness, targetBrightness], runOnStart: false },
     );
 
+    if (homeCockpit === 1) {
```
Comment (ErickSharp) [inline src/instruments/src/ISIS/AutoBrightness.tsx:92]:
```suggestion
    if (homeCockpit) {
```

**Review by ErickSharp — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Settings/Settings.tsx (hunk 5 above)
Comment (ssewell) [reply src/instruments/src/EFB/Settings/Settings.tsx:481]: Was following the pattern set out by the realisticTiller directly above this for the sake of consistency. I'll switch this to a toggle component.

File: src/instruments/src/Common/displayUnit.tsx (hunk 3 above)
Comment (ssewell) [reply src/instruments/src/Common/displayUnit.tsx]: Formatting conflicts with another suggestion. Will manually push another update to address this.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js (hunk 1 above)
Comment (ssewell) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/EICAS_Common.js]: Could you elaborate what you mean by diffed? You mean capture the initial visibility value prior to performing any change?

**Comment by ssewell** [comment]
Currently testing cumulative changes locally.

File: src/instruments/src/EFB/Settings/Settings.tsx (hunk 5 above)
Comment (ssewell) [reply src/instruments/src/EFB/Settings/Settings.tsx]: Toggle component is now used.

**Review by ErickSharp — APPROVED** [review]

**Comment by beheh** [comment]
I could also imagine this new setting taking care of #6475.

**Comment by ssewell** [comment]
> I could also imagine this new setting taking care of #6475, although I guess a throttle might also be present even for casual simmers without a home cockpit.

Good idea. I toyed around with eliminating some of the cockpit sounds (switches, buttons, etc) while keeping desirable cockpit sounds a while back when I was unable to do so using the sim's built-in volume settings. I'll put that on my todo to look into for a potential future enhancement.

File: docs/a320-simvars.md [hunk 7]
```diff
@@ -828,6 +828,14 @@
     - 0 for legacy mode (steering with rudder). 1 for realistic mode with tiller axis
         Tiller axis to be binded on "ENGINE 4 MIXTURE AXIS"
 
+- A32NX_HOME_COCKPIT_ENABLED
+    - Bool
+    - 1 to enable Home Cockpit mode which:
+      - Removes backlight bleed from the PFD, ND, and ECAM displays
+      - Removes reflection from the ISIS
+    - Useful for home cockpits that use the sim's built-in pop-out feature and do not
+        which to have these effects present on their displays.
```
Comment (Benjozork) [inline docs/a320-simvars.md:837]:
```suggestion
    - Useful for home cockpits that use the sim's built-in pop-out feature and do not wish to have these effects present on their displays.
```

**Review by Benjozork — APPROVED** [review]

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #6460
Tier of Testing: 1
Date: 13.1.2022

Testing Process: 
Starting at gate. starting EFB, power up aircraft. 
Checking instruments (popped out)
Change EFB setting
Checking instruments (popped out)

Negatives: none
Not seeing any difference in the popped out windows. 

Testing Results: passed

Media:

https://user-images.githubusercontent.com/16833201/149317177-e9edf087-1021-40a6-9a2c-9257fd4ccd92.mp4

**Comment by mico975** [comment]
Quality Assurance Trainee Report

Discord          : mico#3145
Object of testing: #6460 
Tier of Testing  : 1
Date             : 17/01/2022

Testing Process: 
Started the aircraft and screenshoted all displays.
Enabled "Home cockpit mode"
Again screenshoted all displays.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

Media:       
image
ISIS before
image
ISIS after enabling "Home cockpit mode"
image
Other displays before
image
Other displays after

**Comment by easyjetsimpilot** [comment]
Quality Assurance Tester/Trainee Report

Discord: Easyjetsimpilot
Object of testing: #6460
Tier of Testing: 1
Date: 18/01/22

Testing Process: Compared all screens (at night time for better backlight vision) with Home Cockpit Mode On/Off. Good noticable difference between the two and worked as outlined in the PR. 
Negatives: N/A
Testing Results: Passed

Conclusions: Good PR, simple & effective. Thank you.

**Pull request merged**